hasktorch/src/Torch/NN.hs

```diff
@@ -139,3 +139,34 @@ instance Parameterized Linear
 --     return $ Linear{..}
 
 instance Parameterized [Linear]
+
+data Conv2dSpec = 
+  Conv2dSpec {
+    inputChannelSize  :: Int, 
+    outputChannelSize :: Int, 
+    kernelSize0       :: Int, 
+    kernelSize1       :: Int
+    } deriving (Show, Eq)
+
+data Conv2d = 
+  Conv2d { 
+    conv2dWeight :: Parameter, 
+    conv2dBias   :: Parameter
+    } deriving (Show, Generic)
+
+conv2d'' :: Conv2d -> (Int, Int) -> (Int, Int) -> Tensor -> Tensor
+conv2d'' layer stride padding input = 
+  Torch.Functional.conv2d' w b stride padding input
+    where
+        w = toDependent (conv2dWeight layer)
+        b = toDependent (conv2dBias layer)
+
+instance Randomizable Conv2dSpec Conv2d where
+  sample Conv2dSpec{..} = do
+      w <- makeIndependent =<< randnIO' [outputChannelSize, inputChannelSize, kernelSize0, kernelSize1]
+      b <- makeIndependent =<< randnIO' [outputChannelSize]
+      return $ Conv2d w b
+
+instance Parameterized Conv2d
+
+instance Parameterized [Conv2d]
```

Comment:
I don't think we should be generating this instance FWIW

Reply:
Sure will remove it.

Reply:
I think, for loading and updating parameters of a layer, it is necessary for the layer to be an instance of `Parameterized` class.  Or maybe I am missing something? Feel free to correct me.

Reply:
I think, for loading and updating parameters of an untyped layer, it is necessary for the layer to be an instance of Parameterized class, given my implementation of alexnet or I'll have to modify the implementation a bit. Or maybe I am missing something? Feel free to correct me.

Reply:
I don't see why would you need this instance for `AlexNet`, because you never use a list of `Conv2D`s? Also, if you really need it, then declare the instance in your code and not in the core library.

Reply:
Sorry, I misunderstood I thought you were talking about the `Conv2d` instance. Yeah, it doesn't make sense to have the list instance there.